### PR TITLE
Added new script for existing families that will supersede the gradua…

### PR DESCRIPTION
**Pull request by kdawgsf**
…tion script

### CODE REVIEW

**Comment by kdawgsf** [comment]
@arvindraghavan here is a PR for the student updates/additions for new families. I still need to output another file for address / email address updates, and there are some other TODOs at the top. But, it's ready for testing on an actual destructive import to dp.

File: update_existing_families.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            # Update school
+            studentrecord['SCHOOL'] = district_school_to_dp_school(district_records_dict[student_id]['School'])
+            dp_import_studentrecords.append(studentrecord)
+        elif dp_record['GRADE'] == '8' and dp_record['SCHOOL'] == 'BIS':
+            # Graduation time
+            studentrecord = dp_record.copy()
+            studentrecord['GRADE'] = '9'
+            studentrecord['SCHOOL'] = 'ALUM'
+            dp_import_studentrecords.append(studentrecord)
+
```
Comment (arvindraghavan) [inline update_existing_families.py:95]: Does it make sense to add another branch here for the case where the student is not in the district records and not in 8th grade? Such students have left the district before graduating...

File: update_existing_families.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+for student_id in dp_records_multidict:
+    for dp_record in dp_records_multidict[student_id]:
+        if student_id in district_records_dict:
+            # Returning student
+            studentrecord = dp_record.copy()
+            # Update current grade
+            studentrecord['GRADE'] = district_records_dict[student_id]['Grade']
+            # Update school
+            studentrecord['SCHOOL'] = district_school_to_dp_school(district_records_dict[student_id]['School'])
+            dp_import_studentrecords.append(studentrecord)
```
Comment (arvindraghavan) [inline update_existing_families.py]: Not for this PR, but my sense is that the mailing and e-mail address check goes here, for additional field updates, in the next rev of the script?

File: update_existing_families.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                'FIRST_NAME': siblingrecord['FIRST_NAME'],
+                'LAST_NAME': siblingrecord['LAST_NAME'],
+                'SP_FNAME': siblingrecord['SP_FNAME'],
+                'SP_LNAME': siblingrecord['SP_LNAME'],
+                'ADDRESS': siblingrecord['ADDRESS'],
+                'ADDRESS2': siblingrecord['ADDRESS2'],
+                'ADDRESS3': siblingrecord['ADDRESS3'],
+                'ADDRESS4': siblingrecord['ADDRESS4'],
+                'CITY': siblingrecord['CITY'],
+                'OTHER_ID': '',
```
Comment (arvindraghavan) [inline update_existing_families.py:126]: Except for DONOR_ID, which can come from DP, shouldn't we be taking all other fields from the District data? The assumption here being that the district data for this student (new or not) is the most up-to-date?

File: update_existing_families.py (hunk 1 above)
Comment (kdawgsf) [inline update_existing_families.py:95]: Indeed. Adding.

File: update_existing_families.py (hunk 2 above)
Comment (kdawgsf) [inline update_existing_families.py]: Actually I am thinking that address updates go in a separate output file that does not include 'Other' info. That way we don't need to worry about updating the info for each student in a family etc.

File: update_existing_families.py (hunk 3 above)
Comment (kdawgsf) [inline update_existing_families.py:126]: That will be handled by a second export file that will be imported after this one. This way I can use the existing DP address info in the matching process.

**Comment by kdawgsf** [comment]
@arvindraghavan just pushed another commit with the NOBSD changes. Also pushed a commit with the instructions that were given to us.

File: update_existing_families.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            # Update current grade
+            studentrecord['GRADE'] = district_records_dict[student_id]['Grade']
+            # Update school
+            studentrecord['SCHOOL'] = district_school_to_dp_school(district_records_dict[student_id]['School'])
+        elif dp_record['GRADE'] == '8' and dp_record['SCHOOL'] == 'BIS':
+            studentrecord['GRADE'] = '9'
+            studentrecord['SCHOOL'] = 'ALUM'
+        else:
+            studentrecord['SCHOOL'] = 'NOBSD'
+        if dp_record != studentrecord:
```
Comment (arvindraghavan) [inline update_existing_families.py:93]: Just to be sure, what is your expectation here? The DP record and student record will be the same ONLY if the kid stays in the same grade and school (i.e. repeats the grade), right? (Or am I missing something?) If so, we should probably flag and count occurrences of this corner case. We may also want to count and add a some stats. Probably next PR.

**Comment by arvindraghavan** [comment]
Looks good, let's give this a whirl then. Just to confirm then, this output file will deal with: graduation, grade update, and no-longer-in-district, and adds of students to existing donors. Right?

File: update_existing_families.py (hunk 4 above)
Comment (kdawgsf) [inline update_existing_families.py:93]: The two records will be the same if:
- we run this mid-year
- a student is already NOBSD

I'd like to be able to keep the same process whether we're running a summer update or a mid-year update.

**Comment by kdawgsf** [comment]
Yes, that is right.

**Comment by kdawgsf** [comment]
Looking at the fields that need to be filled out, you'll need to create a new file rather than writing to the file I'm already creating. But the location of the TODO looks right.

**Pull request merged**